Code/Framework/AzCore/AzCore/Math/Sfmt.cpp

```diff
@@ -578,6 +569,8 @@ namespace AZ
     //=========================================================================
     void Sfmt::FillArray64(AZ::u64* array, int size)
     {
+        AZStd::lock_guard<decltype(m_generationMutex)> lock(m_generationMutex);
```

Comment:
repeat: Class template argument deduction can be used here.
```suggestion
        AZStd::scoped_lock lock(m_generationMutex);
```